pkg/docker/docker.go

```diff
@@ -165,3 +166,55 @@ func (c *Client) TarImage(writer io.Writer, image pkgutil.Image) error {
 	}
 	return nil
 }
+
+// ImageListWithFilePath compiles container image names based on all Dockerfiles found, considering excludes
+// according to following search pattern: **/Dockerfile*
+// Return value contains a map with image names and file path
+// Examples for image names with imageName testImage
+// * Dockerfile: `imageName`
+// * sub1/Dockerfile: `imageName-sub1`
+// * sub2/Dockerfile_proxy: `imageName-sub2-proxy`
+func ImageListWithFilePath(imageName string, excludes []string, utils piperutils.FileUtils) (map[string]string, error) {
+
+	imageList := map[string]string{}
+
+	pattern := "**/Dockerfile*"
+
+	matches, err := utils.Glob(pattern)
+	if err != nil || len(matches) == 0 {
```

Comment:
minor: Maybe this function just returns an empty dict if nothing was found and the caller verified the result.